src/components/common/overflow-menu/index.tsx

```diff
@@ -0,0 +1,113 @@
+import Text from '../text';
+import Link from 'next/link';
+import { MdManageAccounts, MdLogout, MdAccountCircle } from 'react-icons/md';
+import { VscGithubAlt } from 'react-icons/vsc';
+import { BiWalletAlt } from 'react-icons/bi';
+import { useState, useRef } from 'react';
+import { cn } from 'utils';
+import Card from '../card';
+const OverflowMenu = props => {
+    const buttonRef = useRef();
+    const [menuOpen, setMenuOpen] = useState(false);
+    // test variables for wallet will be removed later
+    const walletAddress = 'FNfUy8Qp6C9NCD6cz9xHLYSL7n3eFX8LfY1zDx6RcE8G';
+    const walletConnected = false;
+    return (
+        <>
+            <div className="dropdown-end dropdown">
+                <label tabIndex={0}>
+                    <div className="flex flex-row items-center gap-3">
+                        {props.user && (
+                            <Text variant="label">
+                                <Link
+                                    href={`${props.user.user_metadata.user_name}`}
+                                    passHref
+                                >
+                                    {props.user.user_metadata.user_name}
+                                </Link>
+                            </Text>
+                        )}
+                        <button
+                            className={cn(
+                                'flex aspect-square h-fit max-h-full w-fit items-center justify-center gap-3 whitespace-nowrap rounded-full border border-transparent bg-primary p-3 text-black transition-all hover:-translate-y-[0.2rem] hover:bg-white hover:!text-black active:translate-y-[0.05rem] active:scale-95',
+                            )}
+                            onClick={() => {
+                                if (menuOpen) buttonRef.current.blur();
```

Comment:
When cloning the repo, I noticed that after closing the `OverflowMenu`, it will behave erratically once you try to open it again. Removing the `buttonRef.current.blur()` call seems to fix this. Why is the button ref needed again?